malib/algorithm/random/policy.py

```diff
@@ -25,23 +27,33 @@ def __init__(
             custom_config=custom_config,
         )
 
+        self.set_actor(
+            get_model(self.model_config["actor"])(observation_space, action_space)
+        )
+        self.set_critic(
+            get_model(self.model_config["critic"])(
+                observation_space, gym.spaces.Discrete(1)
+            )
+        )
+
     def compute_actions(
         self, observation: DataTransferType, **kwargs
     ) -> DataTransferType:
         raise NotImplementedError
 
     def compute_action(self, observation: DataTransferType, **kwargs) -> Any:
-        actions = range(self.action_space.n)
-        action_prob = torch.zeros(self.action_space.n)
-        # if "legal_moves" in kwargs:
-        #     actions = kwargs["legal_moves"]
-        if "action_mask" in kwargs:
-            actions = np.where(kwargs["action_mask"] == 1)[0]
-            action_prob[actions] = 1.0
-            action_prob /= action_prob.sum()
-        action = np.random.choice(actions)
-
-        return action, None, {"action_probs": action_prob}
+        logits = torch.softmax(self.actor(observation), dim=-1)
+        action_prob = torch.zeros((len(observation), self.action_space.n))
+        if "legal_moves" in kwargs:
+            mask = torch.zeros_like(logits)
+            mask[kwargs["legal_moves"]] = 1
+        elif "action_mask" in kwargs:
+            mask = torch.FloatTensor(kwargs["action_mask"])
+        else:
+            mask = torch.ones_like(logits)
+        logits = mask * logits
+        action = logits.argmax(dim=-1).view((-1, 1)).squeeze(-1).numpy()
```

Comment:
use `argmax` here makes it doesn't act as random policy.
PS: should it also handle for Non-Discrete action space?

Reply:
the random policy is for no-RL-algorithm-involved tests; I think discrete action is enough.